test/patterns/negative/ambiguous.mli

```diff
@@ -0,0 +1,5 @@
+val f : int -> int
+(*@ y = f x
+    ensures match x with
+    | _ | _ when x = 1 -> true
```

Comment:
Arguably this could be accepted, as `x` is not introduced in the guarded pattern.
Your current approximation is good enough to me, though.

Reply:
I thought we choose in #192 to simply disallow such construction in every situation.